concourse/pipelines/model/traits/slack.py

```diff
@@ -32,19 +53,39 @@ def _defaults_dict(self):
 
     def _optional_attributes(self):
         return {
-            'slack_channel',
-            'slack_cfg_name',
+            'default_channel',
         }
 
-    def slack_channel(self):
-        return self.raw.get('slack_channel')
+    def _required_attributes(self):
+        return {
+            'channel_cfgs',
+        }
 
-    def slack_cfg_name(self):
-        return self.raw.get('slack_cfg_name')
+    def _children(self):
+       return self.channel_cfgs().values()
+
+    def channel_cfgs(self):
+        return {
+            name: ChannelConfig(raw_dict=v)
+            for name, v in self.raw.get('channel_cfgs').items()
+        }
+
+    def default_channel(self):
+        return self.raw.get('default_channel')
 
     def transformer(self):
         return SlackTraitTransformer(name=self.name)
 
+    def validate(self):
+        super().validate()
+        default_channel = self.default_channel()
```

Comment:
if this is not a required attribute, this would fail in case no default would be set (because there will probably not be a cfg with 'None' as key